server/jobs/status_sync_job.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package jobs
+
+import (
+	"time"
+
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/mscalendar"
+)
+
+// Unique id for the status sync job
+const statusSyncJobID = "status_sync"
+
+// Run status sync job every 5 minutes
+const statusSyncJobInterval = 5 * time.Minute
+
+// NewStatusSyncJob creates a RegisteredJob with the parameters specific to the StatusSyncJob
+func NewStatusSyncJob(env mscalendar.Env) RegisteredJob {
+	return RegisteredJob{
+		id:                statusSyncJobID,
+		interval:          statusSyncJobInterval,
+		work:              runStatusSyncJob,
+		isEnabledByConfig: isStatusSyncJobEnabled,
+	}
+}
+
+// runStatusSyncJob synchronizes all users' statuses between mscalendar and Mattermost.
+func runStatusSyncJob(env mscalendar.Env) {
+	env.Logger.Debugf("User status sync job beginning")
+
+	_, err := mscalendar.New(env, "").SyncStatusAll()
```

Comment:
0/5: This works, but feels weird in terms of concept. In order to sync the status of all users, it creates a new calendar for the default user (the bot), and that calendar is the one that will sync the status of all the users. That feels weird. Also, SyncStatusAll (as far as I can tell) does not use any functionality of msclaendar, just functionalities of the environment.
Maybe the syncing should be somewhere else?

Reply:
> In order to sync the status of all users, it creates a new calendar for the default user (the bot), and that calendar is the one that will sync the status of all the users.

This is because we use the msgraph application-level permissions to do the "all user" operations, and we must also provide a valid Microsoft user email for the `GetSchedule` API, so the bot also has a dedicated account in the Microsoft Azure instance. Using the application-level permissions, we use batch requests to fetch 400 users' availabilities in one http request.

> Also, SyncStatusAll (as far as I can tell) does not use any functionality of msclaendar, just functionalities of the environment.

I'm not sure what you mean. `SyncStatusAll` calls `m.syncStatusUsers`, which uses the msgraph `GetSchedule` API to gather info on the users.

Reply:
> I'm not sure what you mean. SyncStatusAll calls m.syncStatusUsers, which uses the msgraph GetSchedule API to gather info on the users.
My bad. Not sure how but I did not see that.

With you explanation now all makes sense.